AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -785,6 +785,33 @@ - (NSMutableDictionary *)getReportingParametersForUpdatedRelease:(NSString *)upd
   return reportingParameters;
 }
 
+- (void)changeDistributionGroupIdAfterAppUpdateIfNeeded {
+  NSString *lastDownloadedReleaseHashes = [MS_USER_DEFAULTS objectForKey:kMSDownloadedReleaseHashKey];
+  NSString *lastDownloadedDistributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDownloadedDistributionGroupIdKey];
+  if ((lastDownloadedDistributionGroupId == nil) || ([self isCurrentReleaseWasUpdated:lastDownloadedReleaseHashes] == NO)) {
+    return;
+  }
+  NSString *currentDistributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
+  if ((currentDistributionGroupId != nil) && ([lastDownloadedDistributionGroupId isEqualToString:currentDistributionGroupId])) {
+    return;
+  }
+
+  // Set group ID from downloaded release details.
+  MSLogDebug([MSDistribute logTag], @"Current group ID doesn't match the group ID of downloaded release, updating current group id: %@", lastDownloadedDistributionGroupId);
+  [MS_USER_DEFAULTS setObject:lastDownloadedDistributionGroupId forKey:kMSDistributionGroupIdKey];
+
+  // Remove saved downloaded group ID.
+  [MS_USER_DEFAULTS removeObjectForKey:kMSDownloadedDistributionGroupIdKey];
```

Comment:
Is this temporary used in a same session and remove it? I don't understand why you remove this object.
I feel like you put a value to `NSUserDefaults` and use it in other methods. If this is the case, please do not rely on `NSUserDefaults` and pass the value as a parameter.

Reply:
This value is saved after a new release is downloaded and removed after an app update and restart. After the restart, we check if group ID of downloaded release is the same as current one and update it if not.

I added some code comments so I hope it should be clearer.